src/common/history.c

```diff
@@ -570,31 +584,53 @@ static int _history_copy_and_paste_on_image_merge(int32_t imgid, int32_t dest_im
 
       if(hist)
       {
-        // merge the entry
-        dt_history_merge_module_into_history(dev_dest, dev_src, hist->module, &modules_used, FALSE);
+        double old_iop_order = hist->module->iop_order;
+
+        if (iop_order_version_src != iop_order_version_dest)
+          hist->module->iop_order = dt_ioppr_get_iop_order(dest_iop_list, hist->module->op) + (double)hist->module->multi_priority / 100.0f;
+
+        if (!dt_iop_is_hidden(hist->module))
+        { if (DT_IOP_ORDER_INFO)
+          fprintf(stderr,"\n  module %20s, order %9.5f->%9.5f, multiprio %i",
+                hist->module->op, old_iop_order, hist->module->iop_order, hist->module->multi_priority);
+
+          // merge the entry
+          dt_history_merge_module_into_history(dev_dest, dev_src, hist->module, &modules_used, FALSE);
+        }
       }
 
       l = g_list_previous(l);
     }
   }
   else
   {
+    if (DT_IOP_ORDER_INFO) fprintf(stderr," all modules");
     // we will copy all modules
     GList *modules_src = g_list_first(dev_src->iop);
     while(modules_src)
     {
       dt_iop_module_t *mod_src = (dt_iop_module_t *)(modules_src->data);
 
-      // but only if module is in history in source image
-      if(_search_history_by_module(dev_src, mod_src) != NULL)
+      // only if module is in history of source image, not auto-enabled and not-hidden
+      if((_search_history_by_module(dev_src, mod_src) != NULL) &&
+         !mod_src->default_enabled && !dt_iop_is_hidden(mod_src))
```

Comment:
My suggestion was a little different, by checking also that params != default_params. Also, I just realized, we need to check enabled - some default_enabled modules can be disabled, and that should probably be copy-pastable.

Reply:
Ok, committed as i understood